smart_card_connector_app/src/chrome-api-provider.js

```diff
@@ -355,19 +358,56 @@ function convertConnectionStateToEnum(pcscState) {
   throw new Error(`Unknown connection state value ${state} encountered`);
 }
 
+/**
+ * Generates a debug textual representation of the array of values.
+ *
+ * This function is safe to be used in Release builds, because all potentially
+ * privacy-sensitive data is stripped away from the resulting text.
+ *
+ * For smartCardProviderPrivate API requests and responses all potentially
+ * sensitive data is contained in byte arrays.
+ * @param {!Array} arr
+ * @returns {string}
+ */
+function getArrayDebugRepresentation(arr) {
+  return goog.iter.join(goog.iter.map(arr, (value) => {
+    if ((!goog.DEBUG) && (value instanceof ArrayBuffer)) {
+      return '<stripped value>';
+    }
+    return GSC.DebugDump.dump(value);
+  }), ', ');
+}
+
 /**
  * This class subscribes to a Chrome event with a callback, specified in the
- * constructor arguments. Unsubscribes on disposal.
+ * constructor arguments. Reports the result from the callback to
+ * `reportResultFunction`. Unsubscribes on disposal.
  */
 class ChromeEventListener extends goog.Disposable {
   /**
    * @param {!ChromeBaseEvent<!Function>} chromeEvent
    * @param {!Function} callback
+   * @param {!Function} reportResultFunction
+   * @param {string} nameForLogs
    */
-  constructor(chromeEvent, callback) {
+  constructor(chromeEvent, callback, reportResultFunction, nameForLogs) {
     super();
     this.chromeEvent_ = chromeEvent;
-    this.callback_ = callback;
+    this.callback_ = async (requestId, ...args) => {
+      goog.log.fine(
+          logger,
+          `Processing an event with id ${requestId}: ${nameForLogs}(${
+              getArrayDebugRepresentation(args)})`);
```

Comment:
nit: `getArrayDebugRepresentation(args)` is called twice within the same function, so can be cached in a variable instead.

Also consider doing the same for `getArrayDebugRepresentation(result)`, for readability purposes.

Reply:
done